workflow-bot-app/src/main/java/com/symphony/bdk/workflow/engine/camunda/variable/EscapedJsonVariableDeserializer.java

```diff
@@ -0,0 +1,81 @@
+package com.symphony.bdk.workflow.engine.camunda.variable;
+
+import static com.symphony.bdk.workflow.swadl.v1.Variable.RESOLVED_VALUE_FIELD;
+import static com.symphony.bdk.workflow.swadl.v1.Variable.VARIABLE_REFERENCE_FIELD;
+
+import com.symphony.bdk.workflow.swadl.v1.Variable;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+
+/**
+ * Variables are serialized as escaped JSON by {@link VariableToJsonConverter} and passed to the executor.
+ * When the executor deserializes the activity to access the model, it uses this custom deserializer to
+ * recreate the {@link Variable} objects now containing resolved values.
+ */
+@SuppressWarnings("rawtypes")
+public class EscapedJsonVariableDeserializer extends StdDeserializer<Variable> {
```

Comment:
Shouldn't this be moved to workflow/swadl/ like ActivityDeserializer?

Reply:
this used by CamundaExecutor and relate to how expressions are resolved (so specific to Camunda)